packages/kbn-openapi-generator/src/parser/lib/get_imports_map.ts

```diff
@@ -39,27 +39,46 @@ export const getImportsMap = (parsedSchema: OpenApiDocument): ImportsMap => {
 };
 
 /**
- * Check if an object has a $ref property
+ * Check if an object has a $ref property and return the value
  *
  * @param obj Any object
- * @returns True if the object has a $ref property
+ * @returns The $ref value if it exists
  */
-const hasRef = (obj: unknown): obj is { $ref: string } => {
-  return typeof obj === 'object' && obj !== null && '$ref' in obj;
+const getRef = (obj: unknown): string | undefined => {
+  if (typeof obj === 'object' && obj !== null && '$ref' in obj && typeof obj.$ref === 'string') {
+    return obj.$ref;
+  }
 };
 
+/**
+ * Check if the given reference refers to something not in the same document
+ * # signifies a reference from the root of the document
+ * @param ref $ref value
+ * @returns True if the reference is external
+ */
+const isExternalRef = (ref: string): boolean => !ref.startsWith('#');
+
+/**
+ * Check if the given reference refers to a schema
+ * We do not currently support references to other components
+ * @param ref $ref value
+ * @returns True if the reference is to a schema
+ */
+const isSchemaRef = (ref: string): boolean => ref.includes('#/components/schemas/');
+
 /**
  * Traverse the OpenAPI document recursively and find all references
  *
  * @param obj Any object
  * @returns A list of external references
  */
-function findRefs(obj: unknown): string[] {
+function findExternalSchemaRefs(obj: unknown): string[] {
```

Comment:
We are trying to find refs which will need an import, we also only support schema refs currently so I have made that more explicit.